src/app/server/java/CHIPAppServer-JNI.cpp

```diff
@@ -130,6 +131,12 @@ JNI_METHOD(jboolean, startApp)(JNIEnv * env, jobject self)
     return JNI_TRUE;
 }
 
+JNI_METHOD(jboolean, stopApp)(JNIEnv * env, jobject self)
+{
+    chip::ThreadWork::ChipMainThreadScheduleAndWait([&] { ChipAndroidAppShutdown(); });
```

Comment:
the `&` does not seem require as this does not capture anything.

Either change return value of shutdown to void or add a check + log for errors in this method. Anything returning CHIP_ERROR should be handled (passing it along or logging failures)

Reply:
fixed